### PR TITLE
Foxy migration

### PR DESCRIPTION
**Pull request by pablogs9**
Before merge:

- [x] Create a `foxy` branch and set it as receiving branch
- [x] Merge https://github.com/micro-ROS/rclc/pull/32

### CODE REVIEW

File: rclc/complex_msg_publisher/main.c [hunk 1]
```diff
@@ -1,95 +1,86 @@
...
+
+	// create init_options
+	RCCHECK(rclc_support_init(&support, 0, NULL, &allocator));
+
+	// create node
+	rcl_node_t node = rcl_get_zero_initialized_node();
+	RCCHECK(rclc_node_init_default(&node, "complex_message_node", "", &support));
+
+	// create publisher
+  rcl_publisher_t publisher;
```
Comment (julibert) [inline rclc/complex_msg_publisher/main.c:28]: Indentation.

File: rclc/complex_msg_publisher/main.c [hunk 2]
```diff
@@ -1,95 +1,86 @@
...
+    msg.data14.data4.size = snprintf(msg.data14.data4.data, msg.data14.data4.capacity, "Msg D - %i", num);
+    
     num++;
 
-    ret = rclc_publish(publisher, (const void *)&msg);
+    rcl_ret_t ret = rcl_publish(&publisher, &msg, NULL);
+
     if (ret == RCL_RET_OK) {
-      printf("I send:\n");
+      printf("Sended complex message:\n");
```
Comment (julibert) [inline rclc/complex_msg_publisher/main.c:83]:
```suggestion
      printf("Complex message sent:\n");
```

File: rclc/complex_msg_subscriber/main.c [hunk 3]
```diff
@@ -1,71 +1,92 @@
...
-  printf("\tint64_t: %li\n", msg->data12);
-  printf("\tuint64_t: %lu\n", msg->data13);
-
-  printf("\tstring 1: %s\n", msg->data14.data1.data);
-  printf("\tstring 2: %s\n", msg->data14.data2.data);
-  printf("\tstring 3: %s\n", msg->data14.data3.data);
-  printf("\tstring 4: %s\n", msg->data14.data4.data);
-  printf("\n\n");
+    const complex_msgs__msg__NestedMsgTest * msg = (const complex_msgs__msg__NestedMsgTest *)msgin;
+    printf("Received complex message:\n");
```
Comment (julibert) [inline rclc/complex_msg_subscriber/main.c:17]:
```suggestion
    printf("Complex message received:\n");
```

File: rclc/fragmented_publication/main.c [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+rcl_publisher_t publisher;
+std_msgs__msg__String msg;
+
+void timer_callback(rcl_timer_t * timer, int64_t last_call_time)
+{
+	(void) last_call_time;
+	if (timer != NULL) {
+		RCSOFTCHECK(rcl_publish(&publisher, &msg, NULL));
+		printf("I sent an %ld array\n", msg.data.size);
+
```
Comment (julibert) [inline rclc/fragmented_publication/main.c:25]: Unnecessary line.

File: rclc/package.xml [hunk 5]
```diff
@@ -2,21 +2,27 @@
 <?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
   <name>micro_ros_demos_rclc</name>
-  <version>0.0.1</version>
-  <description>Example of a publisher using int32</description>
-  <maintainer email="julianbermudez@eprosima.com">Julian Bermudez</maintainer>
+  <version>0.0.3</version>
+  <description>Examples using RCL+RCLC API</description>
+  <maintainer email="borjaouterelo@eprosima.com">Borja Outerelo</maintainer>
```
Comment (julibert) [inline rclc/package.xml:7]: Should we set another maintainer?

File: rclc/complex_msg_publisher/main.c (hunk 2 above)
Comment (pablogs9) [reply rclc/complex_msg_publisher/main.c:83]: OMG

File: rclc/complex_msg_publisher/main.c (hunk 1 above)
Comment (pablogs9) [reply rclc/complex_msg_publisher/main.c:28]: Fixed

File: rclc/fragmented_publication/main.c (hunk 4 above)
Comment (pablogs9) [reply rclc/fragmented_publication/main.c:25]: Done

File: rclc/package.xml (hunk 5 above)
Comment (pablogs9) [reply rclc/package.xml:7]: Sure, I'm adding myself and please tell me if you want to add someone else @JanStaschulat @ralph-lange

**Pull request merged**